software/bootcamp-code/image_processor.py

```diff
@@ -104,11 +116,28 @@ def analyze_balls(self, t_balls, fragments) -> list:
             balls.append(Object(x = obj_x, y = obj_y, size = size, distance = obj_dst, exists = True))
 
         balls.sort(key= lambda x: x.distance)
-
-        return balls
+        
+        biggest_ball = Object(exists=False)
+        if len(balls)>0:
+            biggest_ball = balls[-1]
+            
+        return balls, biggest_ball
+    
+    # method for getting the medium depth of the object in the x, y coordinate in 5x3 area around the location
+    def get_depth(self, depth_frame, y, x):
+        center_px=depth_frame[y][x]
+        counter=0
+        sum=0
+        # 5x3 matrix
+        for y_m in range(y, y+4):
```

Comment:
Instead of a for loop you can select a range from the 2d array, then average the values gotten from there. Something like attay[2:4,  2:4] selects a 2x2 area. Also what happens when the point is in the corners?

Reply:
Used numpy mean.